src/sparkbench/src/main/java/org/apache/spark/examples/mllib/JavaKMeans.java

```diff
@@ -87,7 +75,28 @@ public Vector call(Tuple2<LongWritable, VectorWritable> e) {
         }
     });
 
-    KMeansModel model = KMeans.train(points.rdd(), k, iterations, runs, KMeans.K_MEANS_PARALLEL());
+    // Load initial centroids
+    JavaPairRDD<Text, Kluster> clusters = sc.sequenceFile(inputCluster, Text.class, Kluster.class);
+    JavaRDD<Vector> centroids = clusters.map(new Function<Tuple2<Text, Kluster>, Vector>() {
+      @Override
+      public Vector call(Tuple2<Text, Kluster> e) {
+        org.apache.mahout.math.Vector centroid = e._2().getCenter();
+        double[] v = new double[centroid.size()];
+        for (int i = 0; i < centroid.size(); ++i) {
+          v[i] = centroid.get(i);
+        }
+        return Vectors.dense(v);
+      }
+    });
+
+    // Train model
+    KMeansModel initModel = new KMeansModel(centroids.collect());
+    KMeansModel model = new KMeans()
+        .setK(k)
+        .setMaxIterations(iterations)
+        .setRuns(runs)
+        .setInitialModel(initModel)
```

Comment:
Hi @pfxuan , I just got chance to run KMean with `Random` initialization by passing `--initMode Random` in `run.sh`. In my test, the `random` initialization ran much faster and I saw much less stages. The `Parallel` initialization is slow as there are many stages like `collect` and `aggregate` which run several iterations. But these stages are not expected when using `Random` initialization. Can you take a look at the Spark UI to see if there is any difference?